bin/extract_codeobject

```diff
@@ -0,0 +1,426 @@
+#!/usr/bin/perl
+use strict;
+use File::Copy;
+use File::Spec;
+use File::Basename;
+use File::Which;
+use Cwd 'realpath';
+use Getopt::Std;
+use List::Util qw(max);
+
+sub usage {
+  print("Usage: $0 [OPTION]... -i <input>\n");
+  print("Extract the device kernels from an hcc executable.\n\n");
+  print("-h                                \tshow this help message\n");
+  print("-i <input>                        \t[DEPRECATED] input file\n");
+  print("-E <kernel-name>[:<kernel-name>]  \tExtract code object by URI name. Multiple code object names may be specified using \':\' delimiter.\n");
+  print("-L                                \tList code object bundles, by entry name, and their embedded kernel code objects, by URI, available in source input file (-i option).\n");
+  print("-b                                \tList only bundle entries.\n");
+  print("-c <bundle-entry-name>            \tList only embedded code objects for specific code object bundle <bundle-entry-name>\n");
+  print("-o                                \tExtract only binary code objects.  Do not generate disassembled ISA\n");
```

Comment:
I think the default should be to only extract the code objects. The user can always use llvm-obdump to get a disassembly. I think it is better to keep this tool only for extracting.